python/ql/lib/change-notes/2022-05-12-moduleimport-disallow-dots.md

```diff
@@ -0,0 +1,4 @@
+---
+category: fix
+---
+`API::moduleImport` no longer has any results for dotted names, such as `API::moduleImport("foo.bar")`. This previously did not yield usable results, so should not break any existing code.
```

Comment:
Thought it in fact _did_ yield usable results, which is why we're now disallowing it. (In addition to already stating it in the QLDoc.)

Reply:
I'm also a bit confused by this formulation. You mean it had results, but those API nodes were not connected to the graph in the right way?

Reply:
> I'm also a bit confused by this formulation. You mean it had results, but those API nodes were not connected to the graph in the right way?

Yes. That was my point, it had results, but you couldn't really use them for anything. Happy to accept a suggestion to the wording if either of you can make it more clear.

Reply:
The failing tests would seem to disagree with them not being usable for anything. :wink: 

Perhaps we should instead emphasise the real reason why we don't allow `moduleImport("foo.bar.baz")`, that being that it's _less general_ and may miss out on some instances. E.g. it will match

```python
import foo.bar.baz
```

but not 

```python
from foo import bar
bar.baz
```

Reply:
yes, I was actually just working on this (without commenting here :flushed:).